static/js/components/forms/WebsiteCollectionItemForm.tsx

```diff
@@ -32,7 +33,13 @@ export default function WebsiteCollectionItemForm(props: Props): JSX.Element {
       // using plain fetch rather than redux-query here because this
       // use-case doesn't exactly jibe with redux-query: we need to issue
       // a request programmatically on user input.
-      const response = await fetch(url)
+      const response = await debouncedFetch("website-collection", 300, url, {
+        credentials: "include"
+      })
+      if (!response) {
+        // this happens if this fetch was ignored in favor of a later fetch
+        return
+      }
```

Comment:
would it make sense for us to use an [AbortController](https://developer.mozilla.org/en-US/docs/Web/API/AbortController) to cancel requests that are no longer relevant?

Reply:
in addition to debouncing. although it looks like we implemented our own debounce instead of using `_.debounce` or something, so not sure that will play nicely with that

Reply:
I don't think `_.debounce` handles promises and I wanted something that could be awaited on. We can modify the debounce code to allow for aborts, though I don't think we have a UI to let the user request an abort

Reply:
gotcha, that makes sense why we'd need our own in that case